src/Veldrid.Tests/ComputeTests.cs

```diff
@@ -2,23 +2,196 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
+using System.Text;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Veldrid.SPIRV;
 using Xunit;
 
 namespace Veldrid.Tests
 {
     [StructLayout(LayoutKind.Sequential)]
-    public struct BasicComputeTestParams
+    struct FillValueStruct
     {
-        public uint Width;
-        public uint Height;
-        private uint _padding1;
-        private uint _padding2;
+        /// <summary>
+        /// The value we fill the 3d texture with.
+        /// </summary>
+        public float FillValue;
+        public float pad1, pad2, pad3;
+
+        public FillValueStruct(float fillValue)
+        {
+            FillValue = fillValue;
+            pad1 = pad2 = pad3 = 0;
+        }
     }
 
+
     public abstract class ComputeTests<T> : GraphicsDeviceTestBase<T> where T : GraphicsDeviceCreator
     {
+        [Fact]
+        public void ComputeShader3dTexture()
```

Comment:
Added an integration test.  Both the Vulkan and Direct3D11 backends are able to pass the test after the change.  

The OGL backend fails.  I dug into this with renderdoc, and it looked like the 3D texture was getting populated correctly, and the staging textures were getting copied into correctly, but when we use MappableResource, all we get back is a bunch of zeros...

Reply:
There must be a problem with an individual OpenGL codepath. The test succeeds on my setup.